packages/app-staking/src/Overview/MyStake/utils.tsx

```diff
@@ -0,0 +1,240 @@
+    return null;
+  }
+
+  return ledgerOption.unwrapOrDefault().stash.toString();
+}
+
+export async function getStake(
+  stashAccountAddress: string,
+  controllerAccountAddress: string,
+  api: ApiPromise
+): Promise<Stake> {
+  // Get stakeAmount and rewardDestinationsAddresses
+  const stakeLedger = (
+    await api.query.staking.ledger(controllerAccountAddress)
+  ).unwrapOrDefault() as StakingLedger;
+  const stakeAmount = new BN(stakeLedger.total.toString());
+
+  const rewardDestination = await api.query.staking.payee(stashAccountAddress);
+  let rewardDestinationsAddress: string;
+  if (rewardDestination.isStash) {
+    rewardDestinationsAddress = stashAccountAddress;
+  } else if (rewardDestination.isController) {
+    rewardDestinationsAddress = controllerAccountAddress;
+  } else if (rewardDestination.isAccount) {
+    rewardDestinationsAddress = rewardDestination.asAccount.toString();
+  } else {
+    throw new Error(`reward destinations is not stash, controller or account`);
+  }
+
+  // Get nominatorsAddresses
+  const nominates = await getNominates(stashAccountAddress, api);
+
+  return {
+    stashAccountAddress,
+    controllerAccountAddress,
+    stakeAmount,
+    rewardDestinationsAddress,
+    nominates
+  };
+}
+
+export async function getNominates(
+  stashAccountAddress: string,
+  api: ApiPromise
+): Promise<Nominate[]> {
+  const nominates: Nominate[] = [];
+  // Get nominatorsAddresses
+  const nominationsOption = await api.query.staking.nominators(
+    stashAccountAddress
+  );
+  if (nominationsOption.isNone) {
+    return [];
+  }
+  const nominations = nominationsOption.unwrapOrDefault() as Nominations;
+  const nominateToAddresses =
+    nominations && nominations.targets
+      ? (nominations.targets.toJSON() as string[])
+      : [];
+
+  // For each nominator, get stakeShare, commission and nextRewardEstimate
+  await Promise.all(
+    nominateToAddresses.map(
+      async nominateToAddress =>
+        new Promise<void>(async resolve => {
+          // Get commission
+          const commission = (
+            await api.query.staking.validators(nominateToAddress)
+          ).commission.toString();
+
+          // Get stakeShare
+          const { stakeShare, elected } = await getStakeShare(
+            nominateToAddress,
+            stashAccountAddress,
+            api
+          );
+
+          // Get reward estimate
+          const nextRewardEstimate = await getNextRewardEstimate(
+            api,
+            new BN(commission),
+            stakeShare
+          );
+
+          nominates.push({
+            nominateToAddress,
+            stakeShare,
+            commission: new BN(commission),
+            elected,
+            nextRewardEstimate
+          });
+
+          resolve();
+        })
+    )
+  );
+  return nominates;
+}
+
+// Reference: https://github.com/cennznet/cennznet/wiki/Validator-Guide#rewards
+// TODO: is 0.8 dynamic?
+const stakersPayRate = new BN(0.8); // 80% to stakers
```

Comment:
Hi guy, may I double-check, is this stakersPayRate dynamic, if it is, is there any interface I get can this value(I can't find any interface for this)?

Reply:
we plan to build an RPC to return the correct values, in next node update. This value is `100% - validator's commission %` e.g if validator has 10% commission, stakerRate would be 90%

Reply:
Thank you @holygits. I will update the code.